packages/kbn-server-route-repository/src/test_types.ts

```diff
@@ -199,6 +200,12 @@ const repository = {
       });
     },
   }),
+  ...createServerRoute({
+    endpoint: 'POST /internal/endpoint_returning_observable',
+    handler: async () => {
+      return of({ streamed_response: true });
```

Comment:
Do you mean a combination of a KibanaResponse and an Observable? `register_routes` converts an Observable into a response stream using `response.ok({ body: toStream(response) })`

Reply:
Hm, so is there any case where the streamed back result isn't a HTTP 200? What happens in that case, I guess the server throws and a response.customError is sent instead? But how does the client handle that if it expected a stream response?

Reply:
the current expectation is that if a promise resolves, the HTTP status is 200, if it is rejected, we use the status from the rejected value if there is one. Even for an Observable, the handler is async.

Reply:
The challenge I see with this is that this assumption works for HTTP 200 and any errors codes (HTTP 4XX -> 5XX) but it doesn't work for cases that need a different 200 code, like 201 Created for when a resource is created. 

That's why I initially added support for sending back a KibanaReponse from the handler.

But again, I don't know how streaming works, since the headers are sent directly so it'll be a 200 and then errors have to be handled on the client, right?

Reply:
> But again, I don't know how streaming works, since the headers are sent directly so it'll be a 200 and then errors have to be handled on the client, right?

bit of a mix, I don't think it should be one or the other. For instance if the request to get the connector fails, we throw a 400 or whatever. I think, as soon as we encounter a use case for a non-200 for a streaming response, we can fix it.